src/main/kotlin/io/github/gradlenexus/publishplugin/NexusRepository.kt

```diff
@@ -27,11 +27,18 @@ import javax.inject.Inject
 @Suppress("UnstableApiUsage")
 open class NexusRepository @Inject constructor(@Input val name: String, project: Project) {
 
+    @get:Internal
+    internal lateinit var nexusHost: NexusHost
+
```

Comment:
I think this should be a constructor parameter instead.

Reply:
@marcphilipp me too. How would I provide it then? I might have missed that API, but for me it looks like it's not possible to provide the argument when `create` is called.

Reply:
By passing it here like `name` and `project`:
https://github.com/gradle-nexus/publish-plugin/blob/1c7b5099ac276e7a157ea4dfb60b67ee2c2da538/src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishExtension.kt#L55

Reply:
I'm afraid Gradle provides no way to add extra arguments when creating containers with `project.container` API: https://docs.gradle.org/current/javadoc/org/gradle/api/Project.html#container-java.lang.Class-org.gradle.api.NamedDomainObjectFactory-

The factory is hard-coded to have a single name argument only: https://docs.gradle.org/current/javadoc/org/gradle/api/NamedDomainObjectFactory.html

Reply:
Yes, I noticed the same. I was expecting an API like `newInstance`, but couldn't find it. So I thought I might have missed some dark magic in _Gradle_...

Reply:
You're right, I overlooked that little detail... 😬 

Could you please remove the `var` and add a method that configures the urls based on a `NexusHost` parameter?

Reply:
What's wrong with var?
It allows users to modify host after object creation which might be useful

Reply:
@marcphilipp I'm not sure I understand your intent.

In my opinion, `lateinit var nexusHost: NexusHost` would be the most honest approach. Also it fits better into the lazy approach of properties. Evenutally `nexusHost` will only be evaluated when necessary.

Reply:
My intent is to avoid having potentially contradictory properties in `NexusRepository`.

Reply:
The only "contradictory" case I could imagine is when user modifies `nexusUrl` and `snapshotRepositoryUrl` properties.
Do you really mean you do not like that?

If that is the case, the only way to avoid this ambiguity is to make different classes for `OssrhNexusHost` (that is driven by `nexusHost`, and which provides no way to modify `nexusUrl`; and `CustomNexusHost` that lacks `nexusHost` field, and that allows `nexusUrl` customization.
However, I think it would be overkill, and it might be too restrictive for the users

Reply:
However, it might be sligtly better to use `project.objects.property` for `nexusHost`, which would allow clients to configure the host lazily (e.g. based on the value of a Gradle property).

Reply:
> Do you really mean you do not like that?

Yes, I really mean that.

> If that is the case, the only way to avoid this ambiguity is to make different classes for `OssrhNexusHost` (that is driven by `nexusHost`, and which provides no way to modify `nexusUrl`; and `CustomNexusHost` that lacks `nexusHost` field, and that allows `nexusUrl` customization.

We cannot do that without breaking backwards compatibility.

Reply:
@marcphilipp all done :+1: